mezzanine/generic/templatetags/comment_tags.py

```diff
@@ -23,11 +23,11 @@ def comments_for(context, obj):
     """
     form_class = import_dotted_path(settings.COMMENT_FORM_CLASS)
     form = form_class(context["request"], obj)
-    try:
-        context["posted_comment_form"]
-    except KeyError:
+    if context.get("posted_comment_form") is None or \
+                    context.get("posted_comment_form") != form:
```

Comment:
I have tested this and you can see on my site http://www.job360.io/question-answer/cse/how-to-add-two-large-numbers, it's  working perfectly.